src/shims/fs.rs

```diff
@@ -1103,6 +1112,59 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriEvalContextExt<'mir, 'tcx
             this.handle_not_found()
         }
     }
+
+    fn fsync(&mut self, fd_op: OpTy<'tcx, Tag>) -> InterpResult<'tcx, i32> {
+        let this = self.eval_context_mut();
+
+        this.check_no_isolation("fsync")?;
```

Comment:
Rust seems to only use this on Linux, not macos. So maybe we should also restrict the shim to be Linux-only?

Reply:
OTOH, fsync is part of POSIX, so... there's probably no reason to do that. I just wonder why libstd uses another code path on macOS.